php: configure error - your system does not support systemd

**Describe the bug**
Error build php packages.

**To Reproduce**
Steps to reproduce the behavior:
Run `nix-build -I nixpkgs=channel:nixos-unstable-small -A php72.unwrapped`

**Notify maintainers**
@NixOS/php 

Build error occurs after this PR #89388

Build log:
```
checking for select... yes
checking for apparmor... no
checking for sd_notify in -lsystemd-daemon... no
checking systemd/sd-daemon.h usability... yes
checking systemd/sd-daemon.h presence... yes
checking for systemd/sd-daemon.h... yes
configure: error: Your system does not support systemd.
builder for '/nix/store/k3h2hj6vhn93r2v9slbj3nd3fyzsv7ny-php-7.2.29.drv' failed with exit code 1
error: build of '/nix/store/k3h2hj6vhn93r2v9slbj3nd3fyzsv7ny-php-7.2.29.drv' failed

```

--- a/pkgs/development/interpreters/php/default.nix
+++ b/pkgs/development/interpreters/php/default.nix
@@ -195,9 +195,10 @@ let
 
           hardeningDisable = [ "bindnow" ];
 
-          preConfigure = ''
-            # Don't record the configure flags since this causes unnecessary
-            # runtime dependencies
+          preConfigure =
+          # Don't record the configure flags since this causes unnecessary
+          # runtime dependencies
+          ''
             for i in main/build-defs.h.in scripts/php-config.in; do
               substituteInPlace $i \
                 --replace '@CONFIGURE_COMMAND@' '(omitted)' \
@@ -206,7 +207,14 @@ let
             done
 
             export EXTENSION_DIR=$out/lib/php/extensions
-
+          ''
+          # PKG_CONFIG need not be a relative path
+          + lib.optionalString (! lib.versionAtLeast version "7.4") ''
+            for i in $(find . -type f -name "*.m4"); do
+              substituteInPlace $i \
+                --replace 'test -x "$PKG_CONFIG"' 'type -P "$PKG_CONFIG" >/dev/null'
+            done
+          '' + ''
             ./buildconf --copy --force
 
             if test -f $src/genfiles; then

